Backpack/src/main/java/net/skyscanner/backpack/calendar2/view/CalendarHeaderView.kt

```diff
@@ -0,0 +1,83 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018-2021 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.calendar2.view
+
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.util.AttributeSet
+import android.view.Gravity
+import androidx.appcompat.widget.LinearLayoutCompat
+import net.skyscanner.backpack.R
+import net.skyscanner.backpack.calendar2.CalendarParams
+import net.skyscanner.backpack.text.BpkText
+import net.skyscanner.backpack.util.Consumer
+import org.threeten.bp.DayOfWeek
+
+internal class CalendarHeaderView @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyleAttr: Int = 0,
+) : LinearLayoutCompat(context, attrs, defStyleAttr), Consumer<CalendarParams> {
+
+  private val linePaint = Paint(Paint.ANTI_ALIAS_FLAG).apply {
+    style = Paint.Style.FILL_AND_STROKE
+    color = context.getColor(R.color.bpkLine)
+    strokeWidth = resources.getDimension(R.dimen.bpk_calendar_border_width)
+  }
+
+  init {
+    orientation = HORIZONTAL
+  }
+
+  private var lastState: CalendarParams? = null
+
+  override fun invoke(state: CalendarParams) {
+    if (lastState == state) return
+
+    lastState = state
+    removeAllViews()
```

Comment:
this seems quite inefficient to remove and re-add all views whenever the calendar params change - the numbers of days a week are always the same - just the first day might be different, so the views could be added either in the layout or once and then just updated based on the first day and text

Reply:
Done